gap/examples.gi

```diff
@@ -371,6 +371,91 @@ GeneralisedPetersenGraphCons);
 InstallMethod(GeneralisedPetersenGraph, "for integer, integer", [IsInt, IsInt],
 {n, k} -> GeneralisedPetersenGraphCons(IsImmutableDigraph, n, k));
 
+# This function constructs an n by k square grid graph.
+
+InstallMethod(SquareGridGraphCons,
+"for IsMutableDigraph and two positive integers",
+[IsMutableDigraph, IsPosInt, IsPosInt],
+function(filt, n, k)
+  local D, i, j;
+  D := EmptyDigraph(IsMutable, n * k);
+  for i in [1 .. k] do
+    for j in [1 .. (n - 1)] do
+      DigraphAddEdge(D, [((i - 1) * n) + j, ((i - 1) * n) + j + 1]);
+    od;
+  od;
+  for i in [1 .. (k - 1)] do
+    for j in [1 .. n] do
+      DigraphAddEdge(D, [((i - 1) * n) + j, ((i - 1) * n) + j + n]);
+    od;
+  od;
+  return DigraphSymmetricClosure(D);
+end);
+
+InstallMethod(SquareGridGraphCons,
+"for IsImmutableDigraph and two positive integers",
+[IsImmutableDigraph, IsPosInt, IsPosInt],
+function(filt, n, k)
+  local D;
+  D := MakeImmutable(SquareGridGraphCons(IsMutableDigraph, n, k));
+  SetIsMultiDigraph(D, false);
+  SetIsSymmetricDigraph(D, true);
+  return D;
+end);
+
+InstallMethod(SquareGridGraph, "for a function and two positive integers",
+[IsFunction, IsPosInt, IsPosInt],
+SquareGridGraphCons);
+
+InstallMethod(SquareGridGraph, "for two integers", [IsPosInt, IsPosInt],
+{n, k} -> SquareGridGraphCons(IsImmutableDigraph, n, k));
+
+#  This function constructs an n by k triangular grid graph. It is the same as
+#  the square grid graph except that it adds diagonal edges.
+
+InstallMethod(TriangularGridGraphCons,
+"for IsMutableDigraph and two integers",
+[IsMutableDigraph, IsPosInt, IsPosInt],
+function(filt, n, k)
+  local D, i, j;
+  D := EmptyDigraph(IsMutable, n * k);
+  for i in [1 .. k] do
+    for j in [1 .. (n - 1)] do
+      DigraphAddEdge(D, [((i - 1) * n) + j, ((i - 1) * n) + j + 1]);
+    od;
+  od;
+  for i in [1 .. (k - 1)] do
+    for j in [1 .. n] do
+      DigraphAddEdge(D, [((i - 1) * n) + j, ((i - 1) * n) + j + n]);
+    od;
+  od;
```

Comment:
This is just the square grid graph, right? So let's not duplicate code, and just start with that!
```suggestion
  D := SquareGridGraph(IsMutableDigraph, n, k);
```

Reply:
Ah right, that's a good point!